### PR TITLE
Add markdown element

### PR DESCRIPTION
**Pull request by maximehuran**
## Add markdown UI Element

This version will require `twig/extra-bundle` which will generate an error with Sylius Standard setup : 

```
    Twig\Extra\Intl\IntlExtension: ~
```

The recipe will run a script which comment this line : 

https://github.com/monsieurbiz/symfony-recipes/blob/master/monsieurbiz/sylius-rich-editor-plugin/2.9/manifest.json#L10

See https://github.com/Sylius/Sylius-Standard/issues/1021 for more information

### The Ui Element

image

image

## Add highlight JS in shop

image

You can enable or disable and change highlight JS theme : 

```
MONSIEURBIZ_SYLIUS_RICH_EDITOR_ENABLED_HIGHLIGHT_JS_SHOP=true
MONSIEURBIZ_SYLIUS_RICH_EDITOR_ENABLED_HIGHLIGHT_JS_THEME=atom-one-dark
```

No higlight in admin for now.
We have to manage the load after the AJAX call to render elements.

## Custom theme

The default theme is loaded in local.
Another themes are loaded via CDN.

Example with `atom-one-dark`

image

### CODE REVIEW

File: assets/js/highlight.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import hljs from 'highlight.js';
+import 'highlight.js/styles/default.css';
+
+document.addEventListener('DOMContentLoaded', () => {
+  document.querySelectorAll('pre code').forEach((block) => {
```
Comment (jacquesbh) [inline assets/js/highlight.js:5]: Maybe we can be more selective? By adding another class named after the plugin?

**Review by jacquesbh — CHANGES_REQUESTED** [review]

File: src/Twig/RichEditorExtension.php [hunk 2]
```diff
@@ -216,7 +216,7 @@ public function getYoutubeIdFromLink(string $url): ?string
     {
         $isValid = (bool) preg_match(YoutubeUrlValidator::YOUTUBE_REGEX_VALIDATOR, $url, $matches);
 
-        if (!$isValid || !isset($matches[1])) {
+        if (!$isValid) {
```
Comment (maximehuran) [inline src/Twig/RichEditorExtension.php:219]: Spotted by PHP Stan

**Review by delyriand — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -198,6 +218,13 @@ You can distinguish the `Row` element and the `Column` element by their dotted b
 
 ![The HTML element](docs/images/screenshots/html.png)
 
+### Markdown Element
+
+![The Markdown element](docs/images/screenshots/markdown.png)
+
+Use can use `MONSIEURBIZ_SYLIUS_RICH_EDITOR_ENABLED_HIGHLIGHT_JS_SHOP` to enable code highlighting in the shop.
```
Comment (welcoMattic) [inline README.md:225]:
```suggestion
You can use `MONSIEURBIZ_SYLIUS_RICH_EDITOR_ENABLED_HIGHLIGHT_JS_SHOP` to enable code highlighting in the shop.
```

File: scripts/replace_intl_extension.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+function replaceIntlExtensionInTwigConfig($filePath): void
+{
+    $realPath = realpath($filePath);
+
+    if (false === $realPath || !file_exists($realPath)) {
```
Comment (welcoMattic) [inline scripts/replace_intl_extension.php:18]:
```suggestion
    if (!$realPath || !file_exists($realPath)) {
```

**Review by lanfisis — APPROVED** [review]

File: src/Resources/views/Shop/app.html.twig [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{% set highlightJsEnabled = monsieurbiz_sylius_rich_editor.highlight_js.shop_enabled|default('false') %}
+{% set highlightJsTheme = monsieurbiz_sylius_rich_editor.highlight_js.theme|default('default') %}
+
+{% if highlightJsEnabled == 'true' %}
+    {% include '@SyliusUi/_javascripts.html.twig' with {'path': 'bundles/monsieurbizsyliusricheditorplugin/js/rich-editor-highlight.js'} %}
```
Comment (welcoMattic) [inline src/Resources/views/Shop/app.html.twig:5]:
```suggestion
    {{ include('@SyliusUi/_javascripts.html.twig', {'path': 'bundles/monsieurbizsyliusricheditorplugin/js/rich-editor-highlight.js'}) %}
```

Same for the next `include` below

**Pull request merged**